src/engine/init.c

```diff
@@ -1032,7 +1030,7 @@ parse(int argc, char **argv)
 				rc = -DER_INVAL;
 				break;
 			}
-			daos_sysname = optarg;
+			strcpy(daos_sysname, optarg);
```

Comment:
[Nit?] For this to be safe, we have to assume that `optarg` won't be too long. Although that may indeed be true, it's not obvious---we might want to check the length (either by `strlen` or `strncpy`).

Reply:
yes, but if you look at the check right before this strcpy, we already use strnlen for the check and error if it's larger. so this strcpy is safe.

Reply:
I will update it though to use memcpy to avoid coverity warnings though in case it's not smart enough